### ISSUE TITLE
[BUG] System.MissingFieldException

### ISSUE THREAD

**Issue by mysteryx93** [reactiveui/splat#937]
I'm registering this class in my Splat container (also tried `container.Register` and got the same error so it's not the SourceGenerator)

    SplatRegistrations.Register<AboutViewModel>();

When I call `Locator.Current.GetService<AboutViewModel>()`, I get: `System.MissingFieldException`

[![Screenshot-5.png](https://i.postimg.cc/SK64RMcp/Screenshot-5.png)](https://postimg.cc/B8nkwthh)

Base class

```c#
public abstract class AboutViewModelBase<TSettings> : ReactiveObject, IModalDialogViewModel, ICloseable
    where TSettings : SettingsDataBase, new()
{
    private readonly IEnvironmentService _environment;
    private readonly ISettingsProvider<TSettings> _settings;
    private readonly IProcessService _processService;
    private readonly IUpdateService _updateService;

    /// <inheritdoc />
    public event EventHandler? RequestClose;
    /// <inheritdoc />
    public bool? DialogResult { get; } = true;
    
    protected AboutViewModelBase(IEnvironmentService environment, IProcessService processService, ISettingsProvider<TSettings> settings,
        IUpdateService updateService)
    {
        _environment = environment;
        _processService = processService;
        _settings = settings;
        _updateService = updateService;
        // _updateService.FileFormat = UpdateFileFormat;

        // Start in constructor to save time.
        if (!Design.IsDesignMode)
        {
            CheckForUpdates.Execute();
        }
    }
    // rest is properties and commands, pretty standard stuff
```

The one thing that I find strange is that if I hit CTRL+F12 on `base`, it shows 2 classes: one with `ISettingsProvider<TSettings>` and one with `ISettingsProvider<AppSettingsData>`. Could there be a conflict between these 2?

**Comment by mysteryx93** [comment]
I cleared bin/obj folders and restarted the IDE.

I removed "abstract" on the base class.

I tried tweaking the code in various ways, and the only way to make the error go away is to not have any type parameter on the base class...

Even simply adding `AboutViewModelBase<T>` with no reference to T whatsoever makes it crash again!

Does the Splat container not support classes with generic base classes AT ALL?
